src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -435,6 +451,123 @@ export class PatientViewPageStore {
         []
     );
 
+    readonly resourceDefinitions = remoteData({
+        invoke: () =>
+            client.getAllResourceDefinitionsInStudyUsingGET({
+                studyId: this.studyId,
+            }),
+        onResult: defs => {
+            if (defs) {
+                for (const def of defs)
```

Comment:
might want to run this loop in runInAction.   (it's possible onResult is already an action but not sure).

Reply:
looks like onResult is always run inside an action
https://github.com/adufilie/MobxPromise/blob/master/src/MobxPromise.ts#L176